docs(release-notes): [SITE-5722] Solr 9 + D7 beta documentation and release note (#10065)
pantheon-systems/documentation: src/source/content/guides/solr-drupal/04-solr-drupal-7.md

</TabList>

## Solr 9 for Drupal 7 (Beta)

> Apache Solr 9.10.0 support for Drupal 7 is available as a beta release for testing on [Multidev](/guides/multidev) environments. This requires updates to the Pantheon Apache Solr module and your search module. Do not deploy Solr 9 changes to Dev, Test, or Live until GA.

Reviewer: ```suggestion
Apache Solr 9.10.0 support for Drupal 7 is available as a beta release for testing on [Multidev](/guides/multidev) environments. This requires updates to the Pantheon Apache Solr module and your search module. We recommend not deploying Solr 9 changes to Dev, Test, or Live until GA.
```

Solr 9 for Drupal 7 is currently in Beta. Test Solr 9 on a [Multidev](/guides/multidev) environment only. Do not deploy Solr 9 changes to Dev, Test, or Live.

</Alert>

> Drupal 7 sites can now use Apache Solr 9.10.0. This requires updates to the Pantheon Apache Solr module and your search module.

Reviewer: These two statements seem to conflict with each other. "Drupal 7 sites can now use Solr 9 but don't actually do it" is the message we're communicating. Given the oncoming move to GA, is this the guidance we want?

Author: Updated the messaging to make it clear that the release is for beta and needs to be tested with multidev only.

Reviewer: I tweaked the language a bit to be "we recommend waiting" rather than "do not deploy". I feel like if people are ready to deploy Solr 9 to live before the GA, we shouldn't force an arbitrary date if they're already ready to go.

>    During the beta period, merge the drops-7 Solr 9 branch into your site manually. At GA, the updated module will be available as a [one-click upstream update](/core-updates#apply-upstream-updates-via-the-site-dashboard) and these manual steps will no longer be needed.

Reviewer: ```suggestion
   During the beta period, merge the `drops-7` Solr 9 branch into your site manually. At GA, the updated module will be available as a [one-click upstream update](/core-updates#apply-upstream-updates-via-the-site-dashboard) and these manual steps will no longer be needed.
```

1. The updated `pantheon_apachesolr` module with Solr 9 support. Merge the drops-7 Solr 9 branch into your site:

   ```bash
   git checkout -b <multidev-branch>
   git remote add drops-7 git@github.com:pantheon-systems/drops-7.git
   git fetch drops-7
   git merge drops-7/SITE-5456-solr9
   git push origin <multidev-branch>

>    ```

Reviewer: @mehta-asim is this workflow only required during the Beta phase? Do you have a post-launch task to update the docs to remove this if/when it's no longer necessary?

Author: Yes, thats correct, I have updated the alert before the steps and created a new task to clean this up post GA launch
https://getpantheon.atlassian.net/browse/SITE-5742

>    If you encounter merge conflicts while merging this upstream, refer to [Troubleshooting Custom Upstreams](/guides/custom-upstream/troubleshooting) for guidance on resolving them.

Reviewer 2: ```suggestion
   If you encounter merge conflicts while merging this upstream, refer to this [documentation](/guides/custom-upstream/troubleshooting) for guidance on resolving them.
```
